src/python/pants/option/option_value_container.py

```diff
@@ -2,13 +2,48 @@
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
 import copy
+from dataclasses import dataclass
 from typing import Dict, Iterator, List, Optional
 
 from pants.option.ranked_value import Rank, RankedValue, Value
 
 Key = str
 
 
+class OptionValueContainerBuilder:
+    def __init__(self, value_map: Optional[Dict[Key, RankedValue]] = None) -> None:
```

Comment:
I think it is cleaner from an API pov (when using this class in tests for example) to accept this values map as kwarg instead of as an explicit dict. i.e. the code will be 

```python
OptionValueContainerBuilder(
     opt1=RankedValue(value="http://soup.com/api", rank=Rank.CONFIG),
     opt2=RankedValue(value=1, rank=Rank.CONFIG))
```


instead of
```python
OptionValueContainerBuilder(
            dict(
                opt2=RankedValue(value="http://soup.com/api", rank=Rank.CONFIG),
                opt2=RankedValue(value=1, rank=Rank.CONFIG),
            )
        )
```
or at the very least there can be a class method to allow for that, i.e. `OptionValueContainerBuilder.from_kwargs()`

@stuhood @Eric-Arellano